tools/toolchains/__init__.py

```diff
@@ -849,6 +849,21 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    def check_toolchain_path(self, name):
+        """Check if the path to toolchain is valid. Exit if not.
+
+        Positional arguments:
+        name -- the name of the toolchain, as used when creating compile
+                commands to execute. Valid key for TOOLCHAIN_PATHS.
+
+        No return value but causes a system exit of execution if the path
+        does not exist.
+        """
```

Comment:
Do we really want to exit inside of this function? It seems like we'd just want to "check_toolchain_path" here, return True or False, and delegate the appropriate action farther up the call stack. For instance, the toolchains could call this, check the return value, and throw an exception. That could be caught by the CLI scripts (`build.py`, `make.py`, `test.py`, etc) and they could exit gracefully.